src/stan/mcmc/chains.hpp

```diff
@@ -292,6 +291,91 @@ namespace stan {
         return ess;
       }
 
+      /**
+       * Returns the effective sample size for the specified parameter
+       * across all kept samples.
+       *
+       * The implementation is close to the effective sample size
+       * description in BDA3 (p. 286-287).  See more details in Stan
+       * reference manual section "Effective Sample Size".
+       *
+       * Current implementation takes the minimum number of samples
+       * across chains as the number of samples per chain.
+       *
+       * @param std::vector<std::vector<double> > samples
```

Comment:
The parameters and return need textual descriptions.  They don't need to have the type duplicated---that's already in the signature.  

We try to reserve the "sample" for a set of draws and not use "sample" for a single draw.  So this should either be singular "sample" or plural "draws".  Or it could be `chains`.